File: blobs/blob.py
import argparse
import os
import time

ENCODING = "utf-8"
BLOB_MAX = 256

def sync_blobs():
  if os.path.exists("index.json"):
    os.remove("index.json")
  index = open("index.json", "w", encoding=ENCODING)
  index.write("[\n")
  for file in sorted([f.lower() for f in os.listdir('.') if ".blob" in f.lower()]):
    index.write(f"\t\"{file}\",\n")
  index.write("\t\"\"\n]")
  index.close()

def new_blob():
  blob = input("New Blob:\n\n")
  save_new = True
  if len(blob) > BLOB_MAX:
    save_new = False
    continue_save = input(
      f"\n\nINFO: Blobs cannot be more than {BLOB_MAX} characters."
      f" Your blob will be cut to the following:\n\n"
      f"{blob[0:256]}\n\nProceed? (y/n) "
    )
    if continue_save.lower() == "y":
      save_new = True
  if save_new:
    file = open(f"{time.strftime('%Y%m%d%H%M%S')}.blob", "w", encoding=ENCODING)
    file.write(blob[0:256])
    file.close()
    sync_blobs()
    print("\n\nINFO: New blob synced.")
  else:
    print("\n\nINFO: Blob deleted.")

def reset_blobs():
  for file in sorted([f.lower() for f in os.listdir('.') if ".blob" in f.lower()]):
    os.remove(file)
  if os.path.exists("index.json"):
    os.remove("index.json")
  index = open("index.json", "w", encoding=ENCODING)
  index.write("[]")
  index.close()

def main():
  parser = argparse.ArgumentParser(description="Small local blob mangement cli")
  parser.add_argument("-s", "--sync", help="sync your blobs to the index.json file", action="store_true")
  parser.add_argument("-p", "--post", help="compose a new blob", action="store_true")
  parser.add_argument("-r", "--reset", help="reset all blobs", action="store_true")

  args = parser.parse_args()

  if args.sync:
    sync_blobs()
  elif args.post:
    new_blob()
  elif args.reset:
    reset_blobs()
  else:
    print("No args passed. Use -h flag for help.")

if __name__=="__main__":
    main()

File: blobs/blob.lua
-- Help
local HELP_TEXT = [[
usage: blob.lua [-h] [-s] [-p] [-r]

Small local blob management cli

options:
  -h, --help   show this help message and exit
  -s, --sync   sync your blobs to the index.json file
  -p, --post   compose a new blob
  -r, --reset  reset all blobs
]]

-- Settings
local ENCODING = "utf-8"
local BLOB_MAX = 256

local BLOB_PATH = "/path/to/blobs"

local function syncBlobs()
  -- sync the blob index.json file
  os.remove('index.json')
  local index = io.open('index.json', 'w')
  index:write('[\n')
  index:write('\n]')
  index:close()
  local dir = io.open('ls ' ..BLOB_PATH)
  print(dir)
  --[[for name in dir:lines() do
    if (string.sub(name, string.len(name)-5) == ".blob") then
      print(name)
    end
  end]]
end

local function newBlob()
  -- create a new blob
  print("New blob:\n")
  local blob = io.read("*l")
  local save = true
  if (string.len(blob) > BLOB_MAX) then
    save = false
    print("\nINFO: Blobs cannot be more than " .. BLOB_MAX .. " characters. Your blob will be cut to the following:\n") 
    print(string.sub(blob,1,256))
    io.write("\nProceed? (y/n) ")
    io.flush()
    local continue = string.lower(io.read(1))
    if (continue == "y") then
      save = true
    end
  end
  if save then
    local filename = os.date("%Y%m%d%H%M%S") .. ".blob.lua"
    local file = io.open(filename, "w")
    file:write(blob)
    file:close()
    print("\nINFO: New blob saved.")
  else
    print("\nINFO: Blob deleted.")
  end
end

local function isArgUsed(argPassed, shortArg, fullArg)
  if (argPassed == shortArg or argPassed == fullArg) then
   return true
  end
  return false
end

if isArgUsed(arg[1], "-p", "--post") then
  newBlob()
elseif isArgUsed(arg[1], "-s", "--sync") then
  syncBlobs()
elseif isArgUsed(arg[1], "-h", "--help") then
  print(HELP_TEXT)
else
  print("No args passed. Use -h flag for help.")
end
